bin/adhoc-scripts/ParseSpecCmsswdist.py

```diff
@@ -142,22 +142,39 @@ def getDepsSpec(specdir, specfile):
 DEPS_SPEC = []
 
 
-def getDepsRecursive(specdir, specfile):
+def getDepsRecursive(specdir, specfile, buildAllDep=False, recursionDepth=-1):
     """
     - calls getDepsSpec()
     - add all the dependencies that start with `py2-` or `py3-` to deps_spec
     - call recursively getDepsSpec() on all the dependencies
+    :param specdir: The directory holding all the spec files to iterate through
+    :param specfile: The initial spec file.
+    :param buildAllDep: A Bool flag to signal building of all dependencies not only python related
+    :param recursionDepth: An integer to set the maximum recursion depth.
```

Comment:
please describe why value `-1` is used as default value for this parameter

Reply:
hi @vkuznet 
Because the default behavior for the recursive call is - always go to the deepest level possible. It is different only if the recursion depth has been provided through the `-l` parameter or the set to `1` explicitly for the case of `--no-recursion`